### PR TITLE
fix: Receiver_zipkin missing ServiceInstanceID bug #6749

### PR DESCRIPTION
**Pull request by klboke**
fix : https://github.com/apache/skywalking/issues/6749

### CODE REVIEW

File: oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/trace/SpanForward.java [hunk 1]
```diff
@@ -29,15 +30,18 @@
...
 import org.apache.skywalking.oap.server.library.util.BooleanUtils;
+import org.apache.skywalking.oap.server.receiver.zipkin.ZipkinReceiverConfig;
 import org.apache.skywalking.oap.server.receiver.zipkin.handler.SpanEncode;
 import org.apache.skywalking.oap.server.storage.plugin.zipkin.ZipkinSpan;
 import zipkin2.Span;
 import zipkin2.codec.SpanBytesEncoder;
 
 @RequiredArgsConstructor
 public class SpanForward {
+    private static final String DEFAULT_SERVICE_INSTANCE_NAME = "Unknown_instance";
```
Comment (wu-sheng) [inline oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/trace/SpanForward.java:41]:
```suggestion
    private static final String DEFAULT_SERVICE_INSTANCE_NAME = " unknown_instance";
```

File: oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverConfig.java [hunk 2]
```diff
@@ -33,4 +36,5 @@
     private long jettyIdleTimeOut = 30000;
     private int jettyAcceptorPriorityDelta = 0;
     private int jettyAcceptQueueSize = 0;
+    private List<String> instanceNameRule = new ArrayList<>();
```
Comment (wu-sheng) [inline oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverConfig.java:39]: Do OAP core support putting list type?

Comment (klboke) [reply oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverConfig.java:39]: support

Comment (wu-sheng) [reply oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverConfig.java:39]: Could you add a unit test to verify? I can't find the test about this for now. And existing codes use string, and then convert to list.

Comment (wu-sheng) [reply oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverConfig.java:39]: @klboke I can't see the change, why do you resolve this?

File: oap-server/server-storage-plugin/storage-zipkin-elasticsearch7-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/zipkin/elasticsearch/ZipkinTraceQueryEs7DAO.java [hunk 3]
```diff
@@ -211,9 +214,14 @@ public TraceBrief queryBasicTraces(long startSecondTB,
             });
             if (StringUtil.isNotEmpty(serviceId)) {
                 final IDManager.ServiceID.ServiceIDDefinition serviceIDDefinition = IDManager.ServiceID.analysisId(
-                    serviceId);
+                        serviceId);
                 swSpan.setServiceCode(serviceIDDefinition.getName());
             }
+            if (StringUtil.isNotEmpty(serviceInstanceId)) {
```
Comment (wu-sheng) [inline oap-server/server-storage-plugin/storage-zipkin-elasticsearch7-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/zipkin/elasticsearch/ZipkinTraceQueryEs7DAO.java:220]: This seems `serviceInstanceName` still could be NULL.

Comment (wu-sheng) [reply oap-server/server-storage-plugin/storage-zipkin-elasticsearch7-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/zipkin/elasticsearch/ZipkinTraceQueryEs7DAO.java:220]: If you are 100% saying this is not null, why need `if (StringUtil.isNotEmpty(serviceInstanceId))`

Comment (klboke) [reply oap-server/server-storage-plugin/storage-zipkin-elasticsearch7-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/zipkin/elasticsearch/ZipkinTraceQueryEs7DAO.java:220]:
```
type Span {
    traceId: ID!
    segmentId: ID!
    spanId: Int!
    parentSpanId: Int!
    refs: [Ref!]!
    serviceCode: String!
    serviceInstanceName: ID!
    startTime: Long!
    endTime: Long!
    endpointName: String
    # There are three span types: Local, Entry and Exit
    type: String!
    # Peer network id, e.g. host+port, ip+port
    peer: String
    component: String
    isError: Boolean
    # There are 5 layers: Unknown, Database, RPCFramework, Http, MQ and Cache
    layer: String
    tags: [KeyValue!]!
    logs: [LogEntity!]!
}
```
The processing logic here is just to be consistent with the serviceId

File: oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverConfig.java (hunk 2 above)
Comment (klboke) [reply oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverConfig.java:39]: I am going to add a test case. Is there a similar test case for reference?

Comment (klboke) [reply oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverConfig.java:39]: https://github.com/apache/skywalking/pull/6763/commits/0245b00cd3303b6222de9a2d6bd5206986e41c3b

File: oap-server/server-storage-plugin/storage-zipkin-elasticsearch7-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/zipkin/elasticsearch/ZipkinTraceQueryEs7DAO.java (hunk 3 above)
Comment (wu-sheng) [reply oap-server/server-storage-plugin/storage-zipkin-elasticsearch7-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/zipkin/elasticsearch/ZipkinTraceQueryEs7DAO.java:220]: I am saying from the code perspective. What do you want to say to refer the GraphQL?

Comment (klboke) [reply oap-server/server-storage-plugin/storage-zipkin-elasticsearch7-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/zipkin/elasticsearch/ZipkinTraceQueryEs7DAO.java:220]: What I mean is that GraphQL also defines that the serviceCode cannot be empty, but it also makes null processing when processing the serviceId. I am not very clear about the reason, so I refer to the processing logic of the serviceId

File: oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverConfig.java (hunk 2 above)
Comment (wu-sheng) [reply oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverConfig.java:39]: Test case to prove the property with list type working with the current framework.

Comment (klboke) [reply oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverConfig.java:39]: I didn’t understand what you mean, is there any example you can refer to?

Comment (wu-sheng) [reply oap-server/server-receiver-plugin/zipkin-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/zipkin/ZipkinReceiverConfig.java:39]: Take a look at `ModuleManagerTest`.

File: oap-server/server-bootstrap/src/test/java/org/apache/skywalking/oap/server/starter/config/ApplicationConfigLoaderTestCase.java [hunk 4]
```diff
@@ -47,4 +51,13 @@ public void testLoadConfig() {
         Properties properties = (Properties) providerConfig.get("properties");
         assertThat(properties.get("jdbcUrl"), is("jdbc:mysql://localhost:3306/swtest"));
     }
+
+    @Test
+    public void testLoadZipkinReceiverConfig() {
```
Comment (wu-sheng) [inline oap-server/server-bootstrap/src/test/java/org/apache/skywalking/oap/server/starter/config/ApplicationConfigLoaderTestCase.java:56]: You are putting codes in the config test, why name it related to Zipkin? You are testing the `list type` config injection.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**